boards/arm/mm_feather/CMakeLists.txt

```diff
@@ -4,5 +4,10 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 
-zephyr_sources_ifdef(CONFIG_BOOT_FLEXSPI_NOR flexspi_nor_config.c)
-zephyr_sources_ifdef(CONFIG_DEVICE_CONFIGURATION_DATA mmfeather_sdram_ini_dcd.c)
+if(CONFIG_NXP_IMX_RT_BOOT_HEADER)
+  zephyr_compile_definitions(XIP_BOOT_HEADER_ENABLE=1)
+  zephyr_compile_definitions(XIP_BOOT_HEADER_DCD_ENABLE=1)
+  zephyr_compile_definitions(BOARD_FLASH_SIZE=CONFIG_FLASH_SIZE*1024)
```

Comment:
Why were the above 3 compile defines added for this board?

Reply:
If `BOARD_FLASH_SIZE` is not defined, then NXP's HAL will attempt to include `board.h` (which does not exist within Zephyr)

For `XIP_BOOT_HEADER_ENABLE` and `XIP_BOOT_HEADER_DCD_ENABLE`, both are used in `mmfeather_sdram_init_dcd.c`, to control if the DCD header is defined. Without these defines present, the DCD header won't be declared and compilation will fail